### PR TITLE
raftstore: always reject remove leader

### PR DESCRIPTION
**Pull request by BusyJay**
### What is changed and how it works?

What's Changed:

Correctness relies on the assumption that leader won't propose to remove
itself. We used to have a hidden configuration to control the behavior,
this PR make the rejection permanent.

See also #10353.

### Check List 

Tests 

- Unit test
- Integration test

Side effects

- Breaking backward compatibility
    `raftstore.allow-remove-leader` is removed and ignored.

### Release note 

```release-note
Always reject proposing to remove leader
```

### CODE REVIEW

File: components/raftstore/src/store/config.rs [hunk 1]
```diff
@@ -138,8 +138,6 @@ pub struct Config {
     #[config(hidden)]
     pub right_derive_when_split: bool,
 
-    pub allow_remove_leader: bool,
```
Comment (NingLin-P) [inline components/raftstore/src/store/config.rs:141]: Should also remove `allow-remove-leader`

**Review by NingLin-P — COMMENTED** [review]
Rest LGTM

File: components/raftstore/src/store/config.rs (hunk 1 above)
Comment (BusyJay) [reply components/raftstore/src/store/config.rs:141]: Do you mean also search for `allow-remove-leader`? I did, and it's a hidden configuration, so nothing shows up.

Comment (NingLin-P) [reply components/raftstore/src/store/config.rs:141]: It is in some test config file like `components/test_raftstore/src/common-test.toml`

**Review by hicqu — APPROVED** [review]

**Review by gengliqi — APPROVED** [review]

**Comment by BusyJay** [comment]
@gengliqi @hicqu  PTAL, the new change still keeps the configuration for tests and remove for production.

**Review by hicqu — APPROVED** [review]

**Review by gengliqi — APPROVED** [review]

**Pull request merged**